### PR TITLE
[Telegram Action] Add option for Markdown or HTML formatting

### PR DESCRIPTION
**Pull request by memtew**
The Telegram Bot API supports basic formatting for messages. My commit add Markdown formatting mode by default to Telegram action.

### CODE REVIEW

**Comment by memtew** [comment]
But I don't know how to ad an option without breaking current behavior. Maybe through telegram.cfg? My current version of commit doesn't breaks anything.

**Comment by memtew** [comment]
After day of testing I'm found breaking behavior - Markdown parse mode throw error if your message have an alone _ or * symbol.

**Comment by 9037568** [comment]
>But I don't know how to add an option without breaking current behavior. Maybe through telegram.cfg?

Yes, the telegram.cfg would need to be updated with a new key such as `messageFormat` or something similar. Then you'd need to update the action code to fetch the value of the new key and use it appropriately.

**Comment by 9037568** [comment]
>After day of testing I'm found breaking behavior - Markdown parse mode throw error if your message have an alone _ or * symbol.

Post the error and let's see if we can work out how to fix it.

**Comment by memtew** [comment]
> Post the error and let's see if we can work out how to fix it.

Error is simple: if message includes (it doesn't matter where - in variable name like _door_Hall_, in text) unclosed symbols from Markdown (_ or * for example) Telegram api throws error:
`21:50:11.888 [WARN ] [hab.action.telegram.internal.Telegram] - Method failed: HTTP/1.1 400 Bad Request`

**Comment by memtew** [comment]
Added new option to telegram.cfg:
```
#Markdown or HTML
bot.parseMode=Markdown
```
If option not present or blank - Telegram Action just don't use it.

**Comment by 9037568** [comment]
>Added new option to telegram.cfg

When you do this, please make sure to also modify the openhab_default.cfg to reflect the new setting.

**Comment by memtew** [comment]
> In addition to addressing the inline comments, please also run the Eclipse formatter on all source.
Sorry, but I don't have Eclipse installed.

**Comment by 9037568** [comment]
>Sorry, but I don't have Eclipse installed.
Then you'll need to install it or find another way to comply with the formatting requirements.

**Comment by memtew** [comment]
> Then you'll need to install it or find another way to comply with the formatting requirements.

Ok, it was hard, but Eclipse successfully installed on VM. But after Source->Format nothing was changed.

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java [hunk 1]
```diff
@@ -66,10 +66,14 @@
 
     private static Map<String, TelegramBot> groupTokens = new HashMap<String, TelegramBot>();
 
-    public static void addToken(String group, String chatId, String token, String parse) {
-        groupTokens.put(group, new TelegramBot(chatId, token, parse));
+    public static void addToken(String group, String chatId, String token) {
+        groupTokens.put(group, new TelegramBot(chatId, token));
     }
 
+    public static void addToken(String group, String parse) {
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java]: I don't believe this method does anything. The bot created by this method will never function, will it?

**Comment by 9037568** [comment]
>Ok, it was hard, but Eclipse successfully installed on VM. But after Source->Format nothing was changed.

Probably indicating your Eclipse was incorrectly installed. But I can fix the formatting before merging if you can't get it to work.

**Comment by memtew** [comment]
> @9037568 @memtew
> Hi Aydar, hi Chris,
> 
> I've seen this PR and just want to ask, if there is any update on it?
> If not, would it be possible to update the necessary things here to bring up the needed support? (I see only a few lines, which has to be changed, but I'm not familiar with java)
> 
> Best regards,
> Patrick

Hello, Patrick,

My knowledge was not enough to finish the job correctly, so I quit the job. Maybe someone else can do it right. At the moment I use the package that I compiled from the source myself. But in this form, the package does not pass the test of Chris.

**Comment by pschaumburg** [comment]
Hi Aydar,

Thank you for your answer. Maybe @9037568 can give us a short answer :)

Best regards,
Patrick

**Comment by pschaumburg** [comment]
@memtew 
I've done some small thoughts in my own fork with your whole idea and nearly all code in it, but I really don't know, if all of it works. Could you check, if this would work for you?
https://github.com/pschaumburg/openhab1-addons/commit/800c2c5d4d1f94ab59314e27b48ce31ccfbef092
Of if possible, to create a jar file out of it (I will figure out, if all works fine for me).

**Comment by 9037568** [comment]
I believe [this](https://github.com/openhab/openhab1-addons/pull/5490#pullrequestreview-96033478) is the last remaining item, but it's been so long I can't tell for sure.

**Comment by thorstenbe** [comment]
I created a new fork since this is labeled as stale and all issues should be fixed. Formatting works while sending photos as well and I could create a new pull request. Would this be ok?

**Comment by 9037568** [comment]
That'd be fine.

**Comment by thorstenbe** [comment]
Please have a look at #5765

**Comment by 9037568** [comment]
Replaced by #5765

**Pull request closed without merging**